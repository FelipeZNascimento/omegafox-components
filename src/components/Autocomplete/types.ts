export type TDropdownItem = {
  id: number;
  name: string;
  abbreviationEn: string;
};

export interface IAutocompleteProps {
  dropdownList: TDropdownItem[];
  onSelect: (item: TDropdownItem) => void;
}
