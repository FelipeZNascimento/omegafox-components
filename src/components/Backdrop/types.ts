export interface IBackdropProps {
  align?: 'left' | 'center' | 'right';
  children: JSX.Element;
  isOpen: boolean;
}
