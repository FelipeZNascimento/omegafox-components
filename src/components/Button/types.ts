export interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  children?: React.ReactNode | null;
  icon?: React.ReactNode | null;
  isDisabled?: boolean;
  isDarkMode?: boolean;
  isSelected?: boolean;
  leftIcon?: boolean;
  size?: 'small' | 'medium' | 'large';
  shape?: 'rounded' | 'squared';
  shadowed?: boolean;
  tooltip?: string;
  variant?: 'primary' | 'danger' | 'neutral' | 'confirm' | 'warning';
  onClick: () => void;
}
