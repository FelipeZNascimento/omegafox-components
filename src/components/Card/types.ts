export interface ICardProps {
  isForceMobile?: boolean;
  isSelected: boolean;
  title: string;
  subtitle?: string | null;
  onClick?: null | (() => void);
  renderingStatusFunction?: null | (() => JSX.Element | null);
}
