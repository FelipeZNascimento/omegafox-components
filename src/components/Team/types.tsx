export interface ITeamProps {
  id: number;
  align: 'left' | 'right';
  colors: string[];
  isEditable: boolean;
  isForceMobile?: boolean;
  logo: string;
  name: string;
  nameShort: string;
  score: number | null;
}
