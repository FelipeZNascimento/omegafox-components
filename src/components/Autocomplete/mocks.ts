import { TDropdownItem } from './types';

export const dropdownList: TDropdownItem[] = [
  {
    id: 0,
    name: 'Lionel Messi',
    abbreviationEn: 'ARG'
  },
  {
    id: 1,
    name: 'Cristiano Ronaldo',
    abbreviationEn: 'POR'
  },
  {
    id: 2,
    name: 'Mbappé',
    abbreviationEn: 'FRA'
  },
  {
    id: 3,
    name: 'Caio Jr.',
    abbreviationEn: 'BRA'
  },
  {
    id: 4,
    name: 'Darwin Nuñez',
    abbreviationEn: 'URU'
  }
];
