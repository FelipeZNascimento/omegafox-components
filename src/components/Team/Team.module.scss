@import "../../styling/alignments.scss";
@import "../../styling/colors.scss";
@import "../../styling/font-sizes.scss";
@import "../../styling/margins.scss";
@import "../../styling/paddings.scss";

.container {
  @include font-sized(l);
  @include marged(horizontal, s3);

  border-radius: $border-radius;
  flex: 1;
  height: $match-container-height;
  position: relative;
  overflow: hidden;
  z-index: 1;

  &Left {
    @extend .container;
    @include flex-aligned(flex-start, center);
  }
  &Right {
    @extend .container;
    @include flex-aligned(flex-end, center);
  }
}

.logoContainer {
  flex: 1;

  .logo {
    filter: drop-shadow(0px 0px 10px #222);
    position: absolute;
    top: 50%;
    z-index: $hidden-z;
    height: 90px;
    width: auto;

    &Left {
      @extend .logo;
      left: 15%;
      transform: translateX(-50%) translateY(-50%);
    }

    &Right {
      @extend .logo;
      transform: translateX(50%) translateY(-50%);
      right: 15%;
    }
  }
}

.scoreContainer {
  @include bg-rgba-colored($color-black, 0.2);
  @include flex-aligned(center, center);
  @include font-sized(l3);

  height: 100%;
  transition: background-color $transition-time;
  width: 50px;

  &Null {
    @extend .scoreContainer;
    @include bg-rgba-colored($color-black, 0.5);
    transition: background-color $transition-time;
  }
}

.nameContainer {
  @include font-sized(l2);
  @include padded(horizontal, m);
}

.form {
  @include padded(horizontal, s3);
}

input {
  all: unset;
  @include aligned(center);
  border-bottom: 2px solid white;
  width: 36px;
}

input[type="number"]::-webkit-inner-spin-button {
  -webkit-appearance: none;
}
