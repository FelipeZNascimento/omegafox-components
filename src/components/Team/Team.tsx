import { isMobile } from 'react-device-detect';

import { ITeamProps } from './types';
import styles from './Team.module.scss';
import matchLayer from '../../img/match_layer.png';

export const Team = ({
  id,
  align = 'left',
  colors,
  isForceMobile,
  logo,
  name,
  nameShort,
  score
}: ITeamProps) => {
  const renderLeftLogo = () => {
    return (
      <div
        className={styles.containerLeft}
        style={{
          color: colors[0],
          background: `url(${matchLayer}) ${colors[1]}`
        }}
      >
        <div className={styles.logoContainer}>
          <img className={styles.logoLeft} alt="logo" src={logo} />
        </div>
        <div
          className={styles.nameContainer}
          style={{
            textShadow: `-1px 0 ${colors[1]}, 0 1px ${colors[1]}, 1px 0 ${colors[1]}, 0 -1px ${colors[1]}`
          }}
        >
          {isMobile || isForceMobile ? nameShort : name}
        </div>
        <div className={styles.scoreContainer}>{score}</div>
      </div>
    );
  };

  const renderRightLogo = () => {
    return (
      <div
        key={id}
        className={styles.containerRight}
        style={{
          color: colors[0],
          background: `url(${matchLayer}) ${colors[1]}`
        }}
      >
        <div className={styles.scoreContainer}>{score}</div>
        <div
          className={styles.nameContainer}
          style={{
            textShadow: `-1px 0 ${colors[1]}, 0 1px ${colors[1]}, 1px 0 ${colors[1]}, 0 -1px ${colors[1]}`
          }}
        >
          {isMobile || isForceMobile ? nameShort : name}
        </div>
        <div className={styles.logoContainer}>
          <img className={styles.logoRight} alt="logo" src={logo} />
        </div>
      </div>
    );
  };

  return (
    <>
      {align === 'left' && renderLeftLogo()}
      {align === 'right' && renderRightLogo()}
    </>
  );
};
