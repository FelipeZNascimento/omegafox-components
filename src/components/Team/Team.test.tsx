import { render } from '@testing-library/react';

import { Team } from './Team';
import { ITeamProps } from './types';
import { teamLeft } from '../mocks';

describe('TeamComponent', () => {
  const renderComponent = ({
    align,
    colors,
    id,
    logo,
    name,
    nameShort,
    score
  }: ITeamProps) =>
    render(
      <Team
        align={align}
        colors={colors}
        id={id}
        logo={logo}
        name={name}
        nameShort={nameShort}
        score={score}
      />
    );

  it('should render', () => {
    const { container } = renderComponent({
      align: teamLeft.align,
      colors: teamLeft.colors,
      id: teamLeft.id,
      logo: teamLeft.logo,
      name: teamLeft.name,
      nameShort: teamLeft.nameShort,
      score: teamLeft.score
    });

    expect(container).toMatchSnapshot();
  });
});
