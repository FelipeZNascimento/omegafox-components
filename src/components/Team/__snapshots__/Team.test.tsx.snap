// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamComponent should render 1`] = `
<div>
  <div
    class="containerLeft"
    style="color: rgb(25, 51, 117); background: rgb(255, 220, 2) url();"
  >
    <div
      class="logoContainer"
    >
      <img
        alt="Brasil logo"
        class="logo logoLeft logoSmall"
        src="https://assets.omegafox.me/img/countries_crests/bra.png"
      />
    </div>
    <div
      class="nameContainer"
      style="text-shadow: -1px 0 #FFDC02, 0 1px #FFDC02, 1px 0 #FFDC02, 0 -1px #FFDC02;"
    >
      Brasil
    </div>
    <div
      class="scoreContainer"
    >
      4
    </div>
  </div>
</div>
`;
